Automattic/wp-calypso
# Theme Sheet: Turn ThemeDetails data component into a query component, take two

Reboot of #8140 which had to be reverted by #8229 because of #8226

This is mostly motivated as a preparation for #8133.

This PR also:
* Makes the new query component an ES6 class component.
* Introduces a new `isRequesting` type of selector used by that 

To test:
* Make sure that theme sheets still work:
* Land on calypso.localhost:3000/design, click on any theme
* Land on that theme directly (e.g by reloading when already on a theme sheet page). This is especially relevant when logged out to make sure SSR works (try with JS disabled for good measure).
* Also (#8226):
 1. Starting at URL: https://wordpress.com/design (logged-in)
 2. Choose a theme by clicking the large thumbnail/info link (not the ellipsis menu)
 3. Click "Pick this design"
 4. Make sure the theme is activated (as opposed to redirecting to https://wordpress.com/start/with-theme/domains-only)
 5. Try different options and configurations (single site, logged-out, etc) for good measure.
